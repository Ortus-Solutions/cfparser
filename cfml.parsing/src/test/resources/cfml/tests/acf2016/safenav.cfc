component{
 function foo(){
   var xyz={};
   if(xyz?.bar){
   }
 }
}