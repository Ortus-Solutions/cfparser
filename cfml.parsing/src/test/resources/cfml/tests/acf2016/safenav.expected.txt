/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
VAR                      <var>
IDENTIFIER               <xyz>
'='                      <=>
'{'                      <{>
'}'                      <}>
';'                      <;>
Hidden:NEWLINE                  <>
IF                       <if>
'('                      <(>
IDENTIFIER               <xyz>
'?'                      <?>
'.'                      <.>
IDENTIFIER               <bar>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (localAssignmentExpression
                var
                (startExpression
                  (baseExpression (unaryExpression (memberExpression (identifier xyz))))
                )
                =
                (startExpression
                  (baseExpression (unaryExpression (primaryExpression (implicitStruct { }))))
                )
              )
              (endOfStatement ;)
            )
            (statement
              (ifStatement
                if
                (condition
                  (
                  (baseExpression
                    (unaryExpression
                      (memberExpression (identifier xyz) (nullSafeOperator ? .) (identifier bar))
                    )
                  )
                  )
                )
                (statement (compoundStatement { }))
              )
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function foo()     {
      var xyz = {};
      if(xyz.bar )         {

        };

    }
}
/*======*/